src/main/java/net/revelc/code/impsort/ImpSort.java

```diff
@@ -166,4 +178,55 @@ private static void convertAndAddImport(LinkedHashSet<Import> allImports, List<N
     allImports.add(imp);
   }
 
+  /*
+   * Extract all of the tokens from the main body of the file.
+   *
+   * This set of tokens represents all of the file's dependencies, and is used
+   * to figure out whether or not an import is unused.
+   */
+  private static Set<String> extractTokens(CompilationUnit unit) {
+    Set<String> tokens = new LinkedHashSet<String>();
+    NodeList<TypeDeclaration<?>> types = unit.getTypes();
+    types.forEach((Node node) -> {
+      Optional<TokenRange> tokenRange = node.getTokenRange();
+      if (tokenRange.isPresent() && tokenRange.get() != TokenRange.INVALID) {
+        JavaToken token = tokenRange.get().getBegin();
+        while (token != null && token != JavaToken.INVALID) {
+          String tokenString = token.toString();
+          if (!tokenString.isEmpty() && Character.isLetter(tokenString.charAt(0))) {
```

Comment:
Should use `Character.isJavaIdentifierStart` instead of `Characteri.isLetter` (see https://docs.oracle.com/javase/specs/jls/se7/html/jls-3.html#jls-3.8)

Reply:
I was not aware of `Character.isJavaIdentifierStart`, but that fits perfectly. Thanks!